.circleci/config.yml

```diff
@@ -127,57 +132,140 @@ jobs:
+
+      - restore_cache:
+          name: restore mssql installation file
+          key: downloads-{{ checksum "test/bin/install-mssql.ps1" }}
 
       - run:
-          name: build libiconv
-          no_output_timeout: 30m
+          name: setup mssql
           command: |
-            $Env:PATH = "C:\\Ruby<< parameters.ruby_version >>-x64\\bin;$Env:PATH"
-            bundle exec rake ports:libiconv
+            .\test\bin\install-mssql.ps1
+
+      - save_cache:
+          name: save downloads cache
+          paths:
+            - C:\Downloads
+          key: downloads-{{ checksum "test/bin/install-mssql.ps1" }}
 
       - run:
-          name: build freetds
-          no_output_timeout: 30m
+          name: install toxiproxy-server
           command: |
-            $Env:PATH = "C:\\Ruby<< parameters.ruby_version >>-x64\\bin;$Env:PATH"
-            bundle exec rake ports:freetds
+            choco install toxiproxy-server --version=2.5.0 -y
+            Start-Process toxiproxy-server
+
+      - restore_cache:
+          name: restore ports cache
+          keys:
+            - ports-libiconv115-openssl111s-freetds1124
 
       - run:
-          name: build gem
-          no_output_timeout: 30m
+          name: copy ports cache into project
+          command: |
+            Move-Item C:\home\circleci\project\ports .
+
+      - run:
+          name: test gem
           command: |
             $Env:PATH = "C:\\Ruby<< parameters.ruby_version >>-x64\\bin;$Env:PATH"
-            bundle exec rake ports
+            bundle exec rake test
+          environment:
+            TOXIPROXY_HOST: "localhost"
+
+      - store_test_results:
+          path: test/reports
+
+  cross_compile_gem:
+    machine:
+      image: ubuntu-2004:current
+
+    steps:
+      - ruby/install:
+          version: '2.7'
+      - checkout
+      - restore_cache:
+          name: restore gem cache
+          keys:
+            - v1-bundle-{{ .Branch }}-{{ checksum "tiny_tds.gemspec" }}
+            - v1-bundle-{{ .Branch }}-
+            - v1-bundle-
+
+      - run:
+          name: bundle install gems
+          command: |
+            bundle install --path vendor/bundle
+
+      - save_cache:
+          name: save gem cache
+          paths:
+            - ./vendor/bundle
+          key: v1-bundle-{{ .Branch }}-{{ checksum "tiny_tds.gemspec" }}
+
+      - restore_cache:
+          name: restore ports cache
+          keys:
+            - ports-libiconv115-openssl111s-freetds1124
+
+      - run:
+          name: Build gems
+          command: |
+            bundle exec rake gem
+            bundle exec rake gem:native
+
+            # Move gem files to a separate directory, as CircleCI artifacts don't support wildcards for paths
+            mkdir pkg_artifacts
+            mv pkg/*.gem pkg_artifacts
+
+      - store_artifacts:
+          path: pkg_artifacts
```

Comment:
Interesting... ere's some other options for discussion:

1. do this as is
2. store the whole pkg dir (and including the other files besides .gems)
3. run the equivalent of `find ./pkg ! -name "*.gem" -delete` to delete non-.gem files
4. tell `rake gem` to put the .gems somewhere in a folder by themselves

I'm not fully opposed to the current approach, although would you consider calling the new subfolder "gems" since that's what'll go in (rather than pkg_artifacts)?

Also, I recommend having the mkdir/move be a separate job step. Generally my preference is to have more single-purpose job steps rather than job steps with multiple commands.

Reply:
I implemented your suggestions with 741d1c5.